### DISCUSSION TITLE
How to pass args to skopt, like n_jobs for multiprocessing?

### DISCUSSION THREAD

**Discussion by crazy25000** [kernc/backtesting.py#276] [Q&A] [answered]
Hi,

How can we pass arguments to `skopt` when using `method='skopt'` with `backtest.optimize`? Would like to update `n_jobs` because the default is 1.

Tried passing it as a keyword arg, but does not work: `bt.optimize(p1=range(20, 50, 10), p2=range(50, 100, 10), maximize='Profit Factor', constraint=lambda param: param.p1 < param.p2, method='skopt', n_jobs=15)`

https://scikit-optimize.github.io/stable/modules/generated/skopt.optimizer.forest_minimize.html#skopt.optimizer.forest_minimize

> n_jobs int, default: 1
>   The number of jobs to run in parallel for fit and predict. If -1, then the number of jobs is set to the number of cores.

Thank you,

**Comment by kernc** [comment, marked as answer]
> Tried passing it as a keyword arg, but does not work

It doesn't work if it's not implemented in the source.
https://github.com/kernc/backtesting.py/blob/73e1534428fc2db9e681030862133b5067508520/backtesting/backtesting.py#L1247-L1252

Passing `n_jobs != 1` won't help you; `n_jobs=` in scikit-optimize is only used for building the hyperspace landscape estimators, not for running the comparably likely more expensive objective function. Scikit-optimize does _iterative_ model-based optimization, and you get your time back in terms of fewer and more intelligent hyperspace point evaluations.

Don't just believe me; try it for yourself:
https://github.com/kernc/backtesting.py/blob/73e1534428fc2db9e681030862133b5067508520/backtesting/backtesting.py#L1440

**Reply by crazy25000** [reply]
> n_jobs= in scikit-optimize is only used for building the hyperspace landscape estimators, not for running the comparably likely more expensive objective function. 

Oo did not know that. I ran a quick preliminary test earlier, but only looked at number of CPUs used vs. If it actually sped up the computations. I didn't see it in their documentation either. Thanks for the help again!
